pages/requester/create-task.tsx

```diff
@@ -13,12 +13,29 @@ import SecondaryButtonCTA from "../../components/buttons/SecondaryButtonCTA";
 import MCQuestion from "../../components/taskerForm/MCQuestion";
 import LoadingOverlay from "../../components/common/LoadingOverlay";
 import { createTask } from "../../src/Database";
-import { useMoralis } from "react-moralis";
+import { useMoralis, useWeb3ExecuteFunction } from "react-moralis";
 import { useRouter } from "next/router";
 import { computeUnitRewardWei } from "../../src/Helpers";
+import { ethers, BigNumber } from "ethers";
+import EscrowFactory from "../../src/utils/abi/EscrowFactory.json";
+import Escrow from "../../src/utils/abi/Escrow.json";
+import ERC20ABI from "../../src/utils/abi/ERC20Token.json";
+import {
+  StackedLineChartSharp,
+  StackedLineChartTwoTone,
+} from "@mui/icons-material";
 
 export default function Form() {
-  const { isInitialized, Moralis } = useMoralis();
+  const escrowFactoryAddress: string =
+    "0x5dbd6085d4bDCb57eBF5E4b2817D6e20DdEE136b";
+  const maticTokenAddress: string =
+    "0x9c3C9283D3e44854697Cd22D3Faa240Cfb032889";
+  const { isInitialized, user, Moralis } = useMoralis();
+  const requesterAddress = user?.get("ethAddress");
```

Comment:
Referring to `user?` here is unpredictable. I see that you use `requesterAddress` on line 140 - I would put `user.get(...)` (no question mark) directly on that line, and in the surrounding conditional, I would change it to `if (ethereum && user) {...}`.